### PR TITLE
Chore: update dev deps to latest

### PR DESCRIPTION
**Pull request by aladdin-add**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ x] Other, please explain:

#### What changes did you make? (Give an overview)
upgrade all dev deps to latest.

#### Is there anything you'd like reviewers to focus on?
no.

### CODE REVIEW

**Comment by nzakas** [comment]
Looks like we lost this. Do you want to update?

File: .markdownlint.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ MD029: false     # Ordered list item prefix
 MD030: false     # Spaces after list markers
 MD033: false     # Allow inline HTML
 MD034: false     # Bare URL used
```
Comment (nzakas) [inline .markdownlint.yml:18]: This change is unrelated and should be removed from this PR

Comment (bmish) [reply .markdownlint.yml:18]: I'm guessing it's related to updating markdownlint.

**Comment by bmish** [comment]
I'm interested to see this get in but looks like it's pretty out-of-date with master.

File: .markdownlint.yml (hunk 1 above)
Comment (aladdin-add) [reply .markdownlint.yml:18]: yes, it's a new added rule that we were not following.

**Comment by aladdin-add** [comment]
sure, will update later. :)

**Comment by bmish** [comment]
It looks like this is still missing some updates like `glob-parent` (#14879). Any reason that's not included?

**Comment by aladdin-add** [comment]
1. the PR was to upgrade dev deps, but not the deps(`e.g. glob-parent`).
2. some dev deps were not upgraded as breaking changes. e.g. `eslint-plugin-jsdoc` dropped the node v10, but eslint v7 still supports it.

**Review by bmish — APPROVED** [review]

**Comment by bmish** [comment]
@aladdin-add got it, thanks!

File: .markdownlint.yml (hunk 1 above)
Comment (nzakas) [reply .markdownlint.yml:18]: Right. I’m saying such a change doesn’t belong in a PR that is just to upgrade dependencies.

Comment (aladdin-add) [reply .markdownlint.yml]: to not change the config, pushed a commit to fix the md037 issues: https://github.com/eslint/eslint/pull/14624/commits/b8b032a1ee7b573ab245354dcdce5e345cd709f5.

Comment (bmish) [reply .markdownlint.yml]: I would actually recommend just ignoring `CHANGELOG.md` in a `.markdownlintignore` file like this: https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/.markdownlintignore

Comment (aladdin-add) [reply .markdownlint.yml]: sounds good!

Comment (bmish) [reply .markdownlint.yml]: @aladdin-add it looks like this codebase is using markdownlint via CLI and also via its NodeJS API in `Makefile.js`. Only the CLI version will automatically consider the `.markdownlint` file. I think we should probably move this dependency upgrade to a separate PR since it is becoming more complicated. I can take that on if that sounds good.

Comment (aladdin-add) [reply .markdownlint.yml]: agreed to put in another PR! I will revert the markdownlint version.

Comment (bmish) [reply .markdownlint.yml]: Separate PR to update markdownlint: https://github.com/eslint/eslint/pull/14883

**Review by nzakas — APPROVED** [review]

**Comment by snitin315** [comment]
Now `npm run lint` is failing on master with this, we may have to revert this PR.

**Comment by snitin315** [comment]
Strange, it's failing locally only. All good in CI

**Comment by aladdin-add** [comment]
have you reinstalled the deps?

**Comment by snitin315** [comment]
@aladdin-add yes

**Comment by aladdin-add** [comment]
what's your node.js/npm version? maybe you were running on a unwanted version.

**Comment by snitin315** [comment]
Node v14, npm v6.

`rm -rf ./node_modules && npm install` worked for me 👍🏻 , all good now.
